src/node/quote.cpp

```diff
@@ -57,6 +57,57 @@ namespace ccf
     return unique_id;
   }
 
+  std::optional<DigestedPolicy> EnclaveAttestationProvider::get_security_policy_digest(
+    const QuoteInfo& quote_info)
+  {
+    if (access("/dev/sev", F_OK) != 0) {
+      return std::nullopt;
+    }
+
+    DigestedPolicy digest{};
+    CodeDigest d = {};
+    pal::attestation_report_data r = {};
+    try
+    {
+      pal::verify_quote(quote_info, d.data, r);
+      auto quote =
+        *reinterpret_cast<const pal::snp::Attestation*>(quote_info.quote.data());
+      std::copy(
+        std::begin(quote.host_data),
+        std::end(quote.host_data),
+        digest.begin());
+    }
+    catch (const std::exception& e)
+    {
+      LOG_FAIL_FMT("Failed to verify attestation report: {}", e.what());
+      return std::nullopt;
+    }
+
+    return digest;
+  }
+
+  QuoteVerificationResult verify_security_policy_against_store(
+    kv::ReadOnlyTx& tx,
+    const QuoteInfo& quote_info)
+  {
+    if (quote_info.format == QuoteFormat::amd_sev_snp_v1) {
```

Comment:
I'm not sure I understand how this works, it seems to be called with the quote_info.format is virtual, but then checks for sev_snp? Defaulting to verified doesn't seem ideal.